pkg/microservice/aslan/core/common/service/registry/service.go

```diff
@@ -451,3 +457,104 @@ func (s *SwrService) GetImageInfo(option GetRepoImageDetailOption, log *zap.Suga
 
 	return &commonmodels.DeliveryImage{}, nil
 }
+
+type ecrService struct {
+}
+
+func (s *ecrService) getECRService(ep Endpoint, log *zap.SugaredLogger) (*ecr.ECR, error) {
+	creds := credentials.NewStaticCredentials(ep.Ak, ep.Sk, "")
+	config := &aws.Config{
+		Region:      aws.String(ep.Region),
+		Credentials: creds,
+	}
+	sess, err := session.NewSession(config)
+	if err != nil {
+		log.Errorf("Failed to create aws session, err: %s", err)
+		return nil, err
+	}
+	return ecr.New(sess), nil
+}
+
+func (s *ecrService) ListRepoImages(option ListRepoImagesOption, log *zap.SugaredLogger) (resp *ReposResp, err error) {
+	svc, err := s.getECRService(option.Endpoint, log)
+	if err != nil {
+		return nil, err
+	}
+
+	var wg wait.Group
+	var mutex sync.RWMutex
+	resp = &ReposResp{Total: len(option.Repos)}
+	resultChan := make(chan *Repo)
+	defer close(resultChan)
+
+	for _, repo := range option.Repos {
+		name := repo
+		wg.Start(func() {
+			input := &ecr.ListImagesInput{
+				RepositoryName: aws.String(name),
+			}
+			result, err := svc.ListImages(input)
+			if err != nil {
+				log.Errorf("Failed to get image information from aws, error: %s", err)
+				return
```

Comment:
should we return the error for ListRepoImages? Or errors of ListImages will be ignored.